src/engine.cairo

```diff
@@ -110,11 +120,16 @@ pub impl EngineImpl of EngineTrait {
         Opcode::is_opcode_always_illegal(opcode, ref self)?;
 
         if !self.cond_stack.branch_executing() && !flow::is_branching_opcode(opcode) {
+            // check stack size
+            self.check_stack_size()?;
```

Comment:
We wouldn't need to check here if we move the `check_stack_size` to after execute.